gemrb/plugins/GAMImporter/GAMImporter.cpp

```diff
@@ -1136,10 +1136,15 @@ void GAMImporter::PutMazeEntry(DataStream* stream, void* memory) const
 
 int GAMImporter::PutMaze(DataStream* stream, const Game* game) const
 {
-	for (int i = 0; i < MAZE_ENTRY_COUNT; i++) {
-		PutMazeEntry(stream, game->mazedata + i * MAZE_ENTRY_SIZE);
+	if (game->mazedata) {
+		for (int i = 0; i < MAZE_ENTRY_COUNT; i++) {
+			PutMazeEntry(stream, game->mazedata + i * MAZE_ENTRY_SIZE);
+		}
+		PutMazeHeader(stream, game->mazedata + MAZE_ENTRY_COUNT * MAZE_ENTRY_SIZE);
+	} else {
+		stream->WriteFilling(MAZE_DATA_SIZE_HARDCODED);
```

Comment:
This should serve as compatibility to those of our savegames that are empty r/n as thus won't initialize the maze until resave or getting there.